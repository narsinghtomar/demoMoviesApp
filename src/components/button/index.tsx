import React from 'react';
import {
  TouchableOpacity,
  Text,
  View,
  TouchableOpacityProps,
} from 'react-native';
import { styles } from './styles';
import { COLOR_SECONDARY } from '../../utils/colors';
import ButtonLoader from '../buttonloader/index.tsx';

interface ButtonProps extends TouchableOpacityProps {
  buttonStyle?: any
  text: string
  textStyle?: any
  disabled?: boolean
  activeOpacity?: number
  isbuttonloader?: boolean
  testID?: string
}

const Button: React.FC<ButtonProps> = ({
  buttonStyle,
  onPress,
  text,
  textStyle,
  disabled = false,
  activeOpacity = 0.4,
  isbuttonloader = false,
  testID,
}) => {
  const backgroundColorStyle = disabled ? styles.BGDisabled : styles.BGDefault;
  return (
    <TouchableOpacity
      style={[styles.touchableHighlight, backgroundColorStyle, buttonStyle]}
      underlayColor={COLOR_SECONDARY}
      disabled={disabled}
      onPress={onPress}
      activeOpacity={activeOpacity}
      testID={testID}
    >
      <View style={styles.container}>
        <Text style={[styles.text, textStyle]}>{text}</Text>
        {isbuttonloader && <ButtonLoader isbuttonloader={isbuttonloader} />}
      </View>
    </TouchableOpacity>
  );
};

export default Button;
