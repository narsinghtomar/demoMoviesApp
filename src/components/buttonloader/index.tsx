import React, { useEffect } from 'react'
import { View, Animated, Easing } from 'react-native'
import { IMAGES } from '../../assets'
import { styles } from './styles'

interface ButtonLoaderProps {
  isbuttonloader?: boolean
  isRevampLocation?: boolean
  imageName?: any
}

const ButtonLoader: React.FC<ButtonLoaderProps> = ({
  isbuttonloader = false,
  isRevampLocation = false,
  imageName = IMAGES.LOADER,
}) => {
  let rotateValueHolder = new Animated.Value(0)

  useEffect(() => {
    const startImageRotateFunction = () => {
      rotateValueHolder.setValue(0)
      Animated.timing(rotateValueHolder, {
        toValue: 1,
        duration: 1000,
        easing: Easing.linear,
        useNativeDriver: false,
      }).start(() => startImageRotateFunction())
    }
    if (isbuttonloader) {
      startImageRotateFunction()
    }
  }, [isbuttonloader, rotateValueHolder])

  const RotateData = rotateValueHolder.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  })

  return (
    <View style={styles.container}>
      {isbuttonloader && (
        <View style={styles.paddingLeft}>
          <Animated.Image
            resizeMode={'contain'}
            style={[styles.loader, { transform: [{ rotate: RotateData }] }]}
            source={isRevampLocation ? imageName : IMAGES.LOADER}
          />
        </View>
      )}
    </View>
  )
}

export default ButtonLoader
