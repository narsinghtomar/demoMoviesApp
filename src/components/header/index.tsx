import React from 'react';
import {
  SafeAreaView,
  View,
  TouchableOpacity,
  Text,
  Image,
  StyleProp,
  ViewStyle,
  TextStyle,
} from 'react-native';
import { useDispatch } from 'react-redux';
import { IMAGES } from '../../assets';
import { styles } from './styles';
import { resetAuthCompleted } from '../../redux/reducers/authReducer';

interface HeaderProps {
  headerTitle?: string
  headerTextStyle?: StyleProp<TextStyle>
  paddingHorizontal?: number
  isTitleCentered?: boolean
  paddingBottom?: number
  style?: StyleProp<ViewStyle>
  customHeaderContainerStyle?: StyleProp<ViewStyle>
  isShowLogOut?: boolean
  navigation?: any
}

const Header: React.FC<HeaderProps> = ({
  headerTitle,
  headerTextStyle = {},
  paddingHorizontal,
  isTitleCentered = true,
  paddingBottom = 8,
  style,
  customHeaderContainerStyle,
  isShowLogOut = true,
}) => {
  const dispatch = useDispatch();

  const logOut = () => {
    dispatch(resetAuthCompleted());
  };

  return (
    <SafeAreaView style={[styles.safeAreaView, customHeaderContainerStyle]}>
      <View style={[styles.container(paddingHorizontal, paddingBottom), style]}>
        {headerTitle && (
          <Text style={[styles.headerTitle(isTitleCentered), headerTextStyle]}>
            {headerTitle}
          </Text>
        )}
        {isShowLogOut && (
          <TouchableOpacity onPress={logOut} style={styles.touchLogout}>
            <Image
              style={styles.logOutStyle}
              source={IMAGES.LOGOUT}
              resizeMode="contain"
            />
          </TouchableOpacity>
        )}
      </View>
    </SafeAreaView>
  );
};

export default Header;
