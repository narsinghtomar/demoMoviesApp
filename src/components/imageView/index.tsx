import React from 'react';
import {
  View,
  Image,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';
import FastImage from 'react-native-fast-image';
import styles from './styles';

interface ImageViewProps extends TouchableOpacityProps {
  containerstyle?: any
  style?: any
  url?: string
  uri?: any
  priority?: FastImage.priority
  resizeMode?: 'contain' | 'cover'
  isCentered?: boolean
  isClickable?: boolean
  activeOpacity?: number
  onPress?: () => void
}

const ImageView: React.FC<ImageViewProps> = ({
  containerstyle,
  style,
  url,
  uri,
  priority = FastImage.priority.normal,
  resizeMode = 'contain',
  isCentered = false,
  isClickable = false,
  onPress,
  activeOpacity = 0.8,
}) => {
  const imgAlign = isCentered ? 'center' : 'flex-start';
  const ViewWrapper = isClickable ? TouchableOpacity : View;

  return (
    <ViewWrapper
      style={[styles.container(imgAlign), containerstyle]}
      hitSlop={styles.hitSlop}
      onPress={onPress}
      activeOpacity={activeOpacity}
    >
      {url ? (
        <FastImage
          style={[style]}
          source={{
            uri: url,
            priority: priority,
          }}
          resizeMode={resizeMode}
        />
      ) : (
        <Image
          style={[styles.imgStyle, style]}
          source={uri}
          resizeMode={resizeMode}
        />
      )}
    </ViewWrapper>
  );
};

export default ImageView;
