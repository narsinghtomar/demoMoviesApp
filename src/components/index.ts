import Button from './button/index.tsx';
import ButtonLoader from './buttonloader/index.tsx';
import Space from './space/index.tsx';
import InputField from './inputField/index.tsx';
import ImageView from './imageView/index.tsx';
import Header from './header/index.tsx';

export { Button, ButtonLoader, Space, InputField, ImageView, Header };
