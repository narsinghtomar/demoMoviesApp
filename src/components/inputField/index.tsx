import React from 'react'
import {
  TextInput,
  Text,
  View,
  TextInputProps,
  StyleProp,
  ViewStyle,
  TextStyle,
} from 'react-native'
import { COLOR_RED, COLOR_GRAY, COLOR_GRAY_999 } from '../../utils/colors'
import styles from './styles'
import { isRTL } from '../../i18n'

interface InputFieldProps extends TextInputProps {
  textInputName?: string
  inputBoxStyle?: StyleProp<ViewStyle>
  secureInput?: StyleProp<ViewStyle>
  customStyle?: boolean
  validationErrorStyle?: StyleProp<TextStyle>
  isEditable?: boolean | null
  numberOfRow?: number
  labelText?: string
  labelTextStyle?: StyleProp<TextStyle>
  selectionColor?: string
  errorColor?: string
  inputContainer?: StyleProp<ViewStyle>
  textInputRef?: React.RefObject<TextInput>
  errorMessage?: string
  isErrorMsgRequired?: boolean
}

const InputField: React.FC<InputFieldProps> = ({
  placeholder,
  textInputName,
  placeholderTextColor = COLOR_GRAY,
  inputBoxStyle,
  secureInput,
  customStyle = false,
  onChangeText,
  onBlur,
  secureTextEntry,
  validationErrorStyle,
  value,
  isEditable = null,
  numberOfRow,
  labelText,
  keyboardType,
  errorMessage,
  isErrorMsgRequired = false,
  labelTextStyle,
  selectionColor,
  errorColor = COLOR_RED,
  inputContainer,
  textInputRef,
  defaultValue,
  onKeyPress,
  autoCorrect = false,
  autoCapitalize = 'none',
  onSubmitEditing,
  returnKeyType = 'default',
  blurOnSubmit,
}) => {
  const inputFieldStyle = customStyle
    ? secureTextEntry
      ? secureInput
      : inputBoxStyle
    : { ...styles.commonInputBox, ...inputBoxStyle }
  const editableProps = isEditable !== null ? editable : true

  return (
    <View>
      <View style={[styles.textInputContainer, inputContainer]}>
        {labelText && (
          <Text numberOfLines={1} style={[styles.inputLabel, labelTextStyle]}>
            {labelText}
          </Text>
        )}
        <TextInput
          name={textInputName}
          selectionColor={selectionColor}
          placeholder={placeholder}
          placeholderTextColor={placeholderTextColor}
          style={[
            inputFieldStyle,
            {
              borderColor:
                isErrorMsgRequired && errorMessage
                  ? errorColor
                  : COLOR_GRAY_999,
            },
          ]}
          secureTextEntry={secureTextEntry}
          onChangeText={(text) => {
            onChangeText && onChangeText(text) // Added null check
          }}
          onBlur={onBlur}
          value={value}
          editable={editableProps}
          blurOnSubmit={blurOnSubmit}
          numberOfLines={numberOfRow}
          keyboardType={keyboardType}
          ref={textInputRef}
          defaultValue={defaultValue}
          onKeyPress={onKeyPress}
          autoCorrect={autoCorrect}
          autoCapitalize={autoCapitalize}
          onSubmitEditing={onSubmitEditing}
          returnKeyType={returnKeyType}
          textAlign={isRTL() ? 'right' : 'left'}
        />

        {isErrorMsgRequired && errorMessage && (
          <Text style={[styles.error, validationErrorStyle]}>
            {errorMessage}
          </Text>
        )}
      </View>
    </View>
  )
}

export default InputField
