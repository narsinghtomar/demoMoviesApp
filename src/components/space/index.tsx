import React from 'react'
import { View, ViewStyle } from 'react-native'
import { COLOR_TRANSPARENT } from '../../utils/colors'

interface SpaceProps {
  height?: number;
  width?: number;
  backgroundColor?: string;
}

const Space: React.FC<SpaceProps> = ({
  height = 0,
  width = 0,
  backgroundColor = COLOR_TRANSPARENT,
}) => {
  const spaceStyle: ViewStyle = {
    height: height,
    width: width,
    backgroundColor: backgroundColor,
  }

  return <View style={spaceStyle} />
}

export default Space
