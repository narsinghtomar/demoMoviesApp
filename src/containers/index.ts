/**
 * import
 */
import LoginPage from './login';
import MoviesListPage from './moviesList';
/**
 * export
 */
export {LoginPage, MoviesListPage};
