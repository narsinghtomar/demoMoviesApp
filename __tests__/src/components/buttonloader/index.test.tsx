/**
 * ButtonLoader
 */
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { ButtonLoader } from '../../../../src/components'

it('ButtonLoader component renders correctly', () => {
  const tree = renderer.create(<ButtonLoader />).toJSON()
  expect(tree).toMatchSnapshot()
})
