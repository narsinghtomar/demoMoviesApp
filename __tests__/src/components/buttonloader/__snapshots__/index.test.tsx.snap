// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLoader component renders correctly 1`] = `
<View
  style={
    {
      "paddingVertical": 3,
    }
  }
/>
`;
